network/src/commonMain/kotlin/com/wire/kalium/network/KaliumKtorCustomLogging.kt

```diff
@@ -240,6 +249,49 @@ public class KaliumKtorCustomLogging private constructor(
     }
 }
 
+@Suppress("TooGenericExceptionCaught")
+private fun obfuscateAndLogMessage(text: String): String {
+    return try {
+        val obj = (Json.decodeFromString(text) as JsonElement)
+        if (obj.jsonArray.size > 0) {
+            obj.jsonArray.map {
+                obfuscateJsonElement(it)
+            }
+        } else {
+            obfuscateJsonElement(obj)
+        }
+        obj.toString()
+    } catch (e: Exception) {
+        "error the body content while logging "
+    }
+}
+
+fun obfuscateJsonElement(obj: JsonElement) {
+    obj.jsonObject.entries.toMutableSet().map {
+        when {
+            obfuscateJsonKeys.contains(it.key.lowercase()) -> {
+                kaliumLogger.v("${it.key} : ******")
+            }
+            obfuscateJsonIdKeys.contains(it.key.lowercase()) -> {
+                kaliumLogger.v("${it.key} : ${it.value.toString().substring(START_INDEX, END_INDEX)}")
+            }
+            obfuscateObjects.contains(it.key.lowercase()) -> {
+                obfuscateJsonElement(it.value)
+            }
+            else -> {
+                kaliumLogger.e("${it.key} : ${it.value}")
+            }
+        }
+    }
+
+}
+
+private val obfuscateJsonKeys by lazy { listOf("password", "authorization") }
+private val obfuscateJsonIdKeys by lazy { listOf("conversation", "id", "user", "team") }
+private val obfuscateObjects by lazy { listOf("qualified_id") }
```

Comment:
These variables have function names.

```suggestion
private val sensitiveJsonKeys by lazy { listOf("password", "authorization") }
private val sensitiveJsonIdKeys by lazy { listOf("conversation", "id", "user", "team") }
private val sensitiveJsonObjects by lazy { listOf("qualified_id") }
```